File: 53.maximum-subarray.java
// @leet start
class Solution {
    public int maxSubArray(int[] nums) {
		int temp=nums[0];
		int answer=nums[0];
		for (int i = 1; i < nums.length; i++) {
			temp = Math.max(nums[i], nums[i] + temp);
			answer = Math.max(answer, temp);
		}
		return answer;
        
    }
}
// @leet end

File: 53.maximum-subarray.py
# @leet start
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        temp = nums[0]
        answer = nums[0]
        for i in range(1, len(nums)):
            num = nums[i]
            temp = max(num, num + temp)
            answer = max(answer, temp)
        return answer


# @leet end
